routes/articles.js

```diff
@@ -46,11 +46,12 @@ router.post('/article/:id', requireAuth, (req, res) => {
             res.status(404)
             res.end()
         } else {
-            const { title, body, author, color } = results[0]
-            const { newTitle, newBody, newAuthor, newColor } = req.body
+            const { title, body, author, color, published } = results[0]
+            const { newTitle, newBody, newAuthor, newColor, newPublished } = req.body
+            const publishedVal = newPublished === 1 || newPublished === 0 ? newPublished : published
             db.query(
-                'UPDATE articles SET title = ?, body = ?, author = ?, color = ? WHERE id = ?'
-                , [newTitle || title, newBody || body, newAuthor || author, newColor || color, id]
+                'UPDATE articles SET title = ?, body = ?, author = ?, color = ?, published = ? WHERE id = ?'
+                , [newTitle || title, newBody || body, newAuthor || author, newColor || color, publishedVal, id]
```

Comment:
Does toggling between publish and unpublish from the admin side submit an update request and populate unchanged values with old values? If so would it not be better to only update the published column without populating unchanged values again.

Reply:
Yes it would be. This was the simplest solution though, so I went with it.

The reason I like this solution is the method will work with any changes. There are more efficient ways to do it (building an update query based on the input for example) but this was the fastest to implement.